frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -0,0 +1,137 @@
+  let routeData = {
+    show: _.isEmpty(route),
+    create: true,
+    targetPort: _.get(route, 'spec.port.targetPort', ''),
+    unknownTargetPort: '',
+    path: _.get(route, 'spec.path', ''),
+    hostname: _.get(route, 'spec.host', ''),
+    secure: _.has(route, 'spec.termination'),
+    tls: {
+      termination: _.get(route, 'spec.termination', ''),
+      insecureEdgeTerminationPolicy: _.get(route, 'spec.insecureEdgeTerminationPolicy', ''),
+      caCertificate: _.get(route, 'spec.caCertificate', ''),
+      certificate: _.get(route, 'spec.certificate', ''),
+      destinationCACertificate: _.get(route, 'spec.destinationCACertificate', ''),
+      privateKey: _.get(route, 'spec.privateKey', ''),
+    },
+  };
+  if (getResourcesType(resource) === Resources.KnativeService) {
+    const port = _.get(resource, 'spec.template.spec.containers[0].ports[0].containerPort', '');
+    routeData = {
+      ...routeData,
+      unknownTargetPort: _.toString(port),
+    };
+  }
+  return routeData;
+};
+
+export const getBuildData = (buildConfig: K8sResourceKind) => {
+  const buildStrategyType = _.get(buildConfig, 'spec.strategy.type');
+  const buildStrategyData =
+    buildStrategyType === 'Source'
+      ? _.get(buildConfig, 'spec.strategy.sourceStrategy')
+      : _.get(buildConfig, 'spec.strategy.dockerStrategy');
+  const triggers = _.get(buildConfig, 'spec.triggers');
+  const buildData = {
+    env: buildStrategyData.env,
+    triggers: {
+      webhook: checkIfTriggerExists(triggers, 'GitHub'),
+      image: checkIfTriggerExists(triggers, 'ImageChange'),
+      config: checkIfTriggerExists(triggers, 'ConfigChange'),
+    },
+    strategy: buildStrategyType,
+  };
+  return buildData;
+};
+
+export const getServerlessData = (resource: K8sResourceKind) => {
+  let serverlessData = {};
+  if (getResourcesType(resource) === Resources.KnativeService) {
+    const annotations = _.get(resource, 'spec.template.metadata.annotations');
+    serverlessData = {
+      scaling: {
+        minpods: _.get(annotations, 'autoscaling.knative.dev/minScale', 0),
+        maxpods: _.get(annotations, 'autoscaling.knative.dev/maxScale', ''),
+        concurrencytarget: _.get(annotations, 'autoscaling.knative.dev/target', ''),
+        concurrencylimit: _.get(resource, 'spec.template.spec.containerConcurrency', ''),
+      },
+    };
+  }
+  return serverlessData;
+};
+
+export const getDeploymentData = (resource: K8sResourceKind) => {
+  let deploymentData = {};
+  if (getResourcesType(resource) !== Resources.KnativeService) {
+    const containers = _.get(resource, 'spec.template.spec.containers');
+    const triggers = _.get(resource, 'spec.triggers');
+    deploymentData = {
+      env: _.get(containers[0], 'env', []),
+      triggers: {
+        image: checkIfTriggerExists(triggers, 'ImageChange'),
+        config: checkIfTriggerExists(triggers, 'ConfigChange'),
+      },
+      replicas: _.get(resource, 'spec.replicas', 1),
```

Comment:
Why is the default value here `1` for replicas? Any specific reason?

Reply:
Default value is 1 in other import forms as well that is why.